designs/embed_eksd_manifest.md

```diff
@@ -0,0 +1,72 @@
+# Embed EKS-D resources in EKS-A cluster
+
+## Introduction
+
+**Problem:** Currently, the EKS Anywhere controller pulls the EKS Distro manifest from the release URL provided in the versions bundle, which requires access to the internet.
+It would be ideal to have this manifest and its CRDs embedded in the cluster to allow the controller to operate without the requirement to make that external call.
+This will eventually be needed for air-gapped support.
+
+### Goals and Objectives
+
+As an EKS Anywhere user:
+
+* I want the EKSA controller to be able to operate without internet access
+* I want EKSD components to be supported during upgrade
+
+## Overview of Solution
+
+The EKS Distro manifest and the release CRD will be applied directly to the cluster.
+This will allow the controller to grab the EKSD release resources from within the cluster instead of reading in the EKSD manifest from the internet when fetching the applied spec.
+
+### Solution Details
+
+With this feature, the EKSD manifest & its CRD will be applied directly to the cluster.
+We will do this similarly to how we apply the EKSA components to the cluster, introducing a new task to the create workflow called `InstallEksdComponentsTask`.
+We currently don’t pull in the EKSD release CRD in the bundle, so I propose introducing a new field `crdUrl` in the EKSD Bundle to refer to the file in S3 that stores this CRD.
+
+```
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Bundles
+metadata:
+  creationTimestamp: "2022-02-01T16:41:19Z"
+spec:
+  cliMaxVersion: v0.0.0
+  cliMinVersion: v0.0.0
+  number: 1
+  versionsBundles:
+    eksD:
+      channel: 1-21
+      crdUrl: https://distro.eks.amazonaws.com/crds/releases.distro.eks.amazonaws.com-v1alpha1.yaml
+      gitCommit: |
+        fcddf59a516102e20f75b6c672e3aa74cf2da877
+      kindNode:
+        arch:
+```
+
+From here, we can use `kubectl apply` to apply the content from the EKSD release URL and the CRD URL to the cluster.
+
+We will need a way to refer to the EKSD release object on the cluster when calling it from the controller.
+One way to achieve this is by adding a field `EkdsReleaseRef` to the cluster status.
+```
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: test_cluster
+spec:
+  controlPlaneConfiguration:
+    ...
+status:
+  eksdReleaseRef:
+    name: "name of resource" 
+    url: "url we downloaded from"
```

Comment:
I would vote to have the `apiGroup` and `kind` here since the eksd release doesn't belong to the eks-a group.
I would even have the namespace, but if we want to assume it will always be eksa-system, that's ok.

What is the url for? I would vote to remove it of there is no an specific need yet.

Reply:
This will be in the `default` namespace. (only CAPI resources are automatically placed in `eksa-system`)
True, there isnt a specific need for the `url` at the moment. The only thing actually needed is the `name`, so are you proposing the other two fields to be more so informative and show the distinction that it's not in that eks-a group?

Reply:
Why would it be in `default`?
This is not something aimed for users to freely modify. 

From our product perspective, I think we decided that anything that is not part of the api we expose it would be in its specific namespace, so we communicate those are "internal" resources.

From a security perspective, this can lead to problems since it references components of the cluster (like the api server container). So basically having access to update this resource gives an attacker access to decide which are the core components of kubernetes.

Reply:
I agree this should be a part of eksa-system namespace by default, but from the perspective of potentially allowing users to specify it in the future, maybe we should should still ask for a namespace field, and default it to eksa-system.

Url helps determine where we got it from, which I personally think is useful information for a status especially as it is optional. But I am not super set on that so if you feel like it is better not to, that's fine with me